### PR TITLE
Table Injector

### LINKED ISSUE #7: Sigma-Aldrich: Data tables handled incorrectly
Tables are currently not supported in `tungsten`. For example, the hierarchical parser attempts to read tables as hierarchies, and thus nest each cell in a row. For example, in `sigma_aldrich_sial_270989`:
The first row of this table:
image
is read as:
```json
" ... "
                  {
                    "data": "ethyl acetate \n",
                    "children": [
                      {
                        "data": "141-78-6 \n",
                        "children": [
                          {
                            "data": "TWA \n",
                            "children": [
                              {
                                "data": "400 ppm  \n",
                                "children": [
                                  {
                                    "data": "USA. ACGIH Threshold Limit \n",
                                    "children": []
                                  },
                                  {
                                    "data": "Values (TLV) \n",
                                    "children": []
                                  }
                                ]
                              }
                            ]
                          },
" ... "
```
Possible steps to resolution are to create another parsing component dedicated to parsing tables, and creating a separate `GhsSdsItemType` to represent inline tables.

### PR DESCRIPTION
**Pull request by GreenCappuccino**
This PR will close #7 by implementing an injector system for alternative parsers. This parser utilizes `tabula-py` to get tables and their locations.
Original Table:
image
Injector Parse Result:
```
TabulaTable(extraction_method=stream, top=291.0, bottom=491.0, left=80.0, right=557.0, width=477.0, height=200.0, data=
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
|Component        |CAS-No.   |Value |Control parameters|Basis                                                                                  |
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
|diboron trioxide |1303-86-2 |TWA   |10 mg/m3          |USA. ACGIH Threshold Limit Values (TLV)                                                |
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
|diboron trioxide |1303-86-2 |TWA   |10 mg/m3          |USA. NIOSH Recommended Exposure Limits                                                 |
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
|diboron trioxide |1303-86-2 |TWA   |10 mg/m3          |USA. OSHA - TABLE Z-1 Limits for Air Contaminants - 1910.1000                          |
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
|diboron trioxide |1303-86-2 |PEL   |10 mg/m3          |California permissible exposure limits for chemical contaminants (Title 8, Article 107)|
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
|diboron trioxide |1303-86-2 |TWA   |15 mg/m3          |USA. Occupational Exposure Limits (OSHA) - Table Z-1 Limits for Air Contaminants       |
+-----------------+----------+------+------------------+---------------------------------------------------------------------------------------+
, additional_ops=['merge_rows', 'pull_down_to_empty', 'delete_rows_with_empty'])
```
Table data is then injected into the Sigma-Aldrich section hierarchy.

### CODE REVIEW

**Review by Den4200 — CHANGES_REQUESTED** [review]
After a quick glance, I noticed that the tabula jar file is not included in the package. Let's do that!

File: tungsten/__init__.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
     GhsSdsJsonEncoder
-from tungsten.parsers.supplier.sigma_aldrich.sds_parser import \
+)
+from tungsten.parsers.supplier.sigma_aldrich.sds_parser import (
     SigmaAldrichSdsParser
+)
 
 __all__ = ("GhsSdsJsonEncoder", "SigmaAldrichSdsParser")
+
+os.environ["TABULA_JAR"] = "tabula-1.0.6-SNAPSHOT-jar-with-dependencies.jar"
```
Comment (Den4200) [inline tungsten/__init__.py:12]: Relative paths probably won't work here.

Something like this probably will:
```suggestion
os.environ["TABULA_JAR"] = str(Pathlib(__file__).parent / "tabula-1.0.6-SNAPSHOT-jar-with-dependencies.jar").resolve())
```

**Review by Den4200 — CHANGES_REQUESTED** [review]
Relative paths are blasphemous!

File: tungsten/__init__.py (hunk 1 above)
Comment (GreenCappuccino) [reply tungsten/__init__.py]: Implemented in `[057e202](https://github.com/Den4200/tungsten/pull/13/commits/057e2020419da70bef5062db46e274d4e5dff3cd)`

**Review by Den4200 — APPROVED** [review]
LGTM!

**Pull request merged**